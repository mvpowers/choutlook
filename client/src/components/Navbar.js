import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FaTh, FaCog, FaQuestion, FaSkype } from 'react-icons/lib/fa';

const Container = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: #0078d7;
  color: #fff;
`;

const Section = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const Icon = styled.div`
  padding: 15px;

  &:hover {
    background-color: #005a9e;
  }
`;

const Title = styled.div`
  font-size: 17px;
  margin: 0 30px;
`;

const StyledInput = styled.input`
  background-color: #b3d7f3;
  border: 1px solid #0078d7;
  border-radius: 4px;
  padding: 5px;
  color: #333;
`;

const Navbar = ({ password, handleChange }) => (
  <Container>
    <Section>
      <Icon>
        <FaTh size={16} />
      </Icon>
      <Title>Choutlook</Title>
      <StyledInput
        id="password"
        type="text"
        placeholder="Search"
        value={password}
        onChange={handleChange}
      />
    </Section>
    <Section>
      <Icon>
        <FaSkype size={16} />
      </Icon>
      <Icon>
        <FaCog size={16} />
      </Icon>
      <Icon>
        <FaQuestion size={16} />
      </Icon>
    </Section>
  </Container>
);

Navbar.propTypes = {
  password: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default Navbar;
