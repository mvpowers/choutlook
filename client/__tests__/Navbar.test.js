import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Navbar from '../src/components/Message';

Enzyme.configure({ adapter: new Adapter() });

describe('Navbar', () => {
  const handleChange = jest.fn();
  it('should be defined', () => {
    expect(Navbar).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <Navbar password="password" handleChange={handleChange} />
    );
    expect(tree).toMatchSnapshot();
  });
});
