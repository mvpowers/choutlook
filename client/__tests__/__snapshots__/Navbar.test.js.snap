// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    handleChange={[MockFunction]}
    password="password"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          @choutlook.com
        </div>,
        <Message__Text />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            "@choutlook.com",
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          "@choutlook.com",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            @choutlook.com
          </div>,
          <Message__Text />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              "@choutlook.com",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            "@choutlook.com",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
